api/src/main/java/keywhiz/api/automation/v2/CreateClientRequestV2.java

```diff
@@ -0,0 +1,67 @@
+package keywhiz.api.automation.v2;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/**
+ * Model for request to create a new keywhiz client.
+ *
+ * Note: renamed from CreateClientRequest because swagger docs do not handle class name collisions.
+ */
+@AutoValue public abstract class CreateClientRequestV2 {
+  CreateClientRequestV2() {} // prevent sub-classing
+
+  public static Builder builder() {
+    return new AutoValue_CreateClientRequestV2.Builder()
+        .groups()
+        .description(Optional.empty());
+  }
+
+  @AutoValue.Builder public abstract static class Builder {
+    // intended to be package-private
+    abstract Builder groups(ImmutableSet<String> groups);
+    abstract Builder description(Optional<String> description);
+
+    public abstract Builder name(String name);
+
+    public Builder groups(String... groups) {
+      return groups(ImmutableSet.copyOf(groups));
+    }
+
+    public Builder description(String description) {
+      return description(Optional.of(description));
+    }
+
+    public abstract CreateClientRequestV2 build();
+  }
+
+  /**
+   * Static factory method used by Jackson for deserialization
+   */
+  @SuppressWarnings("unused")
+  @JsonCreator public static CreateClientRequestV2 fromParts(
+      @JsonProperty("name") String name,
+      @JsonProperty("groups") Iterable<String> groups,
+      @JsonProperty("description") @Nullable String description) {
+    return builder()
+        .name(name)
+        .groups(ImmutableSet.copyOf(groups))
+        .description(Optional.ofNullable(description))
+        .build();
+  }
+
+  @JsonProperty("name") public abstract String name();
+  @JsonProperty("groups") public abstract ImmutableSet<String> groups();
+  @JsonIgnore public abstract Optional<String> description();
+
+  /* Raw getter for proper API docs generation. */
+  @SuppressWarnings("unused")
+  @JsonProperty("description") @Nullable public String rawDescription() {
+    return description().orElse(null);
```

Comment:
why not return an empty string in the absent case (`orElse("")`) here and then not have the `@Nullable` annotation above?